### ISSUE TITLE
Internal glimmer error when missing component invoked with curlies

### ISSUE THREAD

**Issue by BryanCrotaz** [glimmerjs/glimmer-vm#1183]
Bug in Ember 3.22
```
Cannot read property 'getCapabilities' of undefined@ 40 ms
Source:     
TypeError: Cannot read property 'getCapabilities' of undefined
    at Object.evaluate (http://192.168.0.10:7357/assets/vendor.js:50753:52)
    at AppendOpcodes.evaluate (http://192.168.0.10:7357/assets/vendor.js:49084:19)
    at LowLevelVM.evaluateSyscall (http://192.168.0.10:7357/assets/vendor.js:52502:22)
    at LowLevelVM.evaluateInner (http://192.168.0.10:7357/assets/vendor.js:52458:14)
    at LowLevelVM.evaluateOuter (http://192.168.0.10:7357/assets/vendor.js:52450:14)
    at JitVM.next (http://192.168.0.10:7357/assets/vendor.js:53505:24)
    at JitVM.execute (http://192.168.0.10:7357/assets/vendor.js:53477:25)
    at TryOpcode.handleException (http://192.168.0.10:7357/assets/vendor.js:52623:23)
    at UpdatingVMFrame.handleException (http://192.168.0.10:7357/assets/vendor.js:52857:31)
    at UpdatingVM.throw (http://192.168.0.10:7357/assets/vendor.js:52554:18)
```
it happens when you refer to a curly-component that doesn't exist in a test template - changing it to angle brackets gives a more useful message `Cannot find component my-missing-component`. Only happens when you're passing attributes.
```
await render (hbs`{{my-missing-component}}`); // gives a useful missing component error
await render (hbs`{{my-missing-component select=false}}`); // gives the cryptic glimmer error
await render (hbs`<MyMissingComponent @select={{false}}/>`); // gives a useful missing component error
```

**Comment by rwjblue** [comment]
Thanks for reporting this! This definitely sucks, we need to fix ASAP.

@wycats / @pzuraq / @krisselden - Y'all have any idea of what is going on here?

**Comment by lifeart** [comment]
Same for missing modifiers

**Comment by pzuraq** [comment]
@lifeart can you provide more details?

**Comment by lifeart** [comment]
@pzuraq discovered while debugging https://github.com/lifeart/ember-ref-bucket/issues/11

If I has typo in modifier name inside modifier test, I see same issue

**Comment by lifeart** [comment]
![image](https://user-images.githubusercontent.com/1360552/97469977-0b293b00-1958-11eb-941c-6122fb4dcf68.png)
![image](https://user-images.githubusercontent.com/1360552/97470008-12e8df80-1958-11eb-8187-2df9215da807.png)
@pzuraq  ^

**Comment by lifeart** [comment]
![image](https://user-images.githubusercontent.com/1360552/97470121-37dd5280-1958-11eb-8c12-3940a5cbda89.png)
definition don't have manager property

**Comment by lifeart** [comment]
may be partially related to https://github.com/glimmerjs/glimmer-vm/issues/1173

**Comment by pzuraq** [comment]
@lifeart yes I'm pretty sure I know what the problem is for components specifically and I believe that there is a fix we can implement in Ember. It is not related to #1173, which does throw an error correctly, with an unclear error message.

#1171 should fix this and #1173 more permanently, as resolution is much more unified in that refactor, and all resolution errors throw a reasonable, readable message.

**Comment by lifeart** [comment]
@pzuraq looks like issue bubbling from outlet or such, and it appear only if modifier has incorrect name :/

![image](https://user-images.githubusercontent.com/1360552/97471951-56444d80-195a-11eb-947a-2139604a4685.png)

**Comment by pzuraq** [comment]
These appear to be separate issues, I read the original bug report in lifeart/ember-ref-bucket#11 and that appears very different from these other cases.

Each one of these ultimately comes down to the fact that we don't have a uniform system for resolution in general. That's what #1171 moves us toward.

**Comment by lifeart** [comment]
@lifeart tracking down ref-bucket case with getCapabilities error, pointing me to https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/syntax/outlet.ts#L101

Is it possible what component manager is not assigned to helper/component if some modifier not found? (some tricky error or such)

**Comment by pzuraq** [comment]
These error messages should be better on master now, as we have a much more unified resolution system in general. Please let me know if not and reopen this issue!

**Comment by meirish** [comment]
Hello!  I just ran into this (my stack track was exactly like the image in this comment: https://github.com/glimmerjs/glimmer-vm/issues/1183#issuecomment-718083128) in the upgrade to Ember 3.24, so I'm guessing the changes haven't been upstreamed, but I was wondering what the best way track this getting back into Ember is - is there a particular commit / PR to watch?

thanks!

**Comment by pzuraq** [comment]
These changes should have been integrated, Ember.js lags 6 to 12 weeks behind changes in Glimmer VM due to the beta release cycle. You can look out for Glimmer VM PRs in Ember.js, and try out using Ember Canary or betas to see if the issue has been fixed before the next version is released. I believe these issues should be addressed in 3.25, in particular.

**Comment by knownasilya** [comment]
Seeing this in glimmer-experimental

```
TypeError: Cannot read properties of undefined (reading 'getCapabilities')
    at ConstantsImpl.component (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/program/dist/commonjs/es2017/lib/constants.js:163:68)
    at renderInvocation (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/runtime/dist/commonjs/es2017/lib/render.js:77:40)
    at Object.renderComponent (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/runtime/dist/commonjs/es2017/lib/render.js:115:10)
    at Object.getTemplateIterator (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/core/dist/commonjs/src/render-component/index.js:67:29)
    at renderToStream (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/ssr/dist/commonjs/src/render.js:50:38)
    at /Users/ilya/maintained/spark-poc/node_modules/@glimmer/ssr/dist/commonjs/src/render.js:40:9
    at new Promise (<anonymous>)
    at Object.renderToString (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/ssr/dist/commonjs/src/render.js:34:12)
    at Proxy.renderToString (/Users/ilya/maintained/spark-poc/node_modules/@glimmerx/ssr/dist/commonjs/src/render.js:8:18)
    at render (/Users/ilya/maintained/spark-poc/src/entry-server.ts:6:41)
TypeError: Cannot read properties of undefined (reading 'getCapabilities')
    at ConstantsImpl.component (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/program/dist/commonjs/es2017/lib/constants.js:163:68)
    at renderInvocation (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/runtime/dist/commonjs/es2017/lib/render.js:77:40)
    at Object.renderComponent (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/runtime/dist/commonjs/es2017/lib/render.js:115:10)
    at Object.getTemplateIterator (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/core/dist/commonjs/src/render-component/index.js:67:29)
    at renderToStream (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/ssr/dist/commonjs/src/render.js:50:38)
    at /Users/ilya/maintained/spark-poc/node_modules/@glimmer/ssr/dist/commonjs/src/render.js:40:9
    at new Promise (<anonymous>)
    at Object.renderToString (/Users/ilya/maintained/spark-poc/node_modules/@glimmer/ssr/dist/commonjs/src/render.js:34:12)
    at Proxy.renderToString (/Users/ilya/maintained/spark-poc/node_modules/@glimmerx/ssr/dist/commonjs/src/render.js:8:18)
    at render (/Users/ilya/maintained/spark-poc/src/entry-server.ts:6:41)
```
